[DOC] Add MirrorMaker property descriptions to appendix

PaulRMellor opened the pull request:
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Moved the property descriptions from the Kafka MirrorMaker configuration and  Kafka MirrorMaker 2.0 configuration sections
MM and MM2 procedure callouts link to the appropriate sections
Not every property is described for MM in the appendix, just covering the properties that were described in the main part of the guide.

Part of a work in progress to move all property descriptions into one place.

- Removed 14 files
- Additional property description content added through documentation/api/*.adoc files, which are included and built with the appendix
- New common properties section for all components: modules/con-common-configuration-proprieties.adoc
- Removed the :context: settings defined in the mirrormaker config assemblies, as they're no longer needed - no replication.
- Moved the full descriptions of JVM options and Container Images into the appendix. Non-MM components retain a small section that link to these, until they have procedure callouts to link from. 
- Moved the Kafka Connect logging section into appendix (for MM2), again retaining a small section.
- Made formats consistent for api adoc files (.title to describe a property)

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

scholzj commented on documentation/modules/con-common-configuration-properties.adoc:
```diff
@@ -0,0 +1,394 @@
+When using Kafka with a Kafka cluster not managed by Strimzi, you can specify the bootstrap servers list according to the configuration of the given cluster.
+
+[id='con-common-configuration-cipher-{context}']
+.`SSL` cipher suites
```
This does not describe only cipher suite configuration. So maybe it should be named _SSL configuration`?

PaulRMellor replied:
It's all _configuration_. Trying to use the property names in the title, so maybe .`ssl` is enough.

scholzj replied:
Then it should either just SSL or maybe _SSL protocol versions and cipher suites_

PaulRMellor replied:
I've left at _ssl_

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMaker2ClusterSpec.adoc:
```diff
@@ -1,18 +1,9 @@
+
+Standard Apache Kafka configuration may be provided, xref:assembly-kafka-connect-configuration-deployment-configuration-kafka-connect[restricted to those properties not managed directly by Strimzi].
+
+To run external listeners with a specific _cipher suite_ for a TLS version, you can xref:con-common-configuration-cipher-reference[configure allowed `ssl` properties].
```
This does not configure listeners but clients.

PaulRMellor replied:
I changed to _For client connection using a specific cipher suite_ and assumed we needed this change wherever else this sentence was used.

scholzj commented on documentation/modules/con-common-configuration-properties.adoc:
```diff
@@ -0,0 +1,394 @@
+.Example SSL configuration
+[source,yaml,subs="attributes+"]
+----
+spec:
+  producer:
+    config:
+      ssl.cipher.suites: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384" <1>
+      ssl.enabled.protocols: "TLSv1.2" <2>
+      ssl.protocol: "TLSv1.2" <3>
```
Thsi shows up in many different variants ... I wonder if it should be just

```suggestion
# ...
config:
  ssl.cipher.suites: "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384" <1>
  ssl.enabled.protocols: "TLSv1.2" <2>
  ssl.protocol: "TLSv1.2" <3>
# ...
```

in the example.

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc:
```diff
@@ -1,18 +1,65 @@
+
+* Kafka cluster bootstrap address
+* Security (encryption, authentication, and authorization)
+* Consumer group identifier
```
We should add _Interceptor classes_ to the list

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerConsumerSpec.adoc:
```diff
@@ -1,18 +1,65 @@
+* `sasl.`
+* `security.`
+* `bootstrap.servers`
+* `group.id`
```
We should add `interceptor.classes` to the list

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc:
```diff
@@ -1,18 +1,50 @@
+* `sasl.`
+* `security.`
+* `bootstrap.servers`
+* `group.id`
```
We should add `interceptor.classes` to the list and remove `group.id` which is consumer only.

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerProducerSpec.adoc:
```diff
@@ -1,18 +1,50 @@
+However, there are exceptions for options automatically configured and managed directly by Strimzi related to:
+
+* Kafka cluster bootstrap address
+* Security (encryption, authentication, and authorization)
```
We should add Interceptor classes to the list

scholzj commented on documentation/api/io.strimzi.api.kafka.model.KafkaMirrorMakerTls.adoc:
```diff
@@ -1,5 +1,8 @@
+
+.`trustedCertificates`
+Having xref:type-KafkaClientAuthenticationTls-reference[set `tls` to configure TLS encryption], use the
+`trustedCertificates` property to provide a list of secrets with key names under which the certificates are stored in X.509 format.
```
Can we add something like this at the end?

```
If you want to enable TLS but use the default set of public certification authorities shipped with Java, you can specify the `trustedCertificates` as empty array:

.An example of enabling TLS with the default Java certificates 
[source,yaml,subs=attributes+]
----
tls:
  trustedCertificates: []
----
```

PaulRMellor replied:
Added

scholzj commented on documentation/api/io.strimzi.api.kafka.model.authentication.KafkaClientAuthenticationOAuth.adoc:
```diff
@@ -44,11 +42,11 @@ authentication:
 You can configure the access token used for authentication with the Kafka broker directly.
 In this case, you do not specify the `tokenEndpointUri`.
 In the `accessToken` property, specify a link to a `Secret` containing the access token.
 
+
```
Do we need the 2 empty lines here?

scholzj commented on documentation/modules/con-common-configuration-properties.adoc:
```diff
@@ -0,0 +1,394 @@
+* `KafkaTopic` uses a replication factor to configure the number of replicas of each partition within a Kafka cluster.
+* Kafka components use replicas to configure the number of pods in a deployment to provide better availability and scalability.
+
+*MirrorMaker*
+
+When running Kafka MirrorMaker on Kubernetes it is not absolutely necessary to run multiple replicas of the Kafka MirrorMaker for high availability.
+When the node where the Kafka MirrorMaker has deployed crashes, Kubernetes will automatically reschedule the Kafka MirrorMaker pod to a different node.
+However, running Kafka MirrorMaker with multiple replicas can provide faster failover times as the other nodes will be up and running.
```
This applies to also for example Kafka Connect. But it seems a bit weird to mention it explicitely in section called _Common configuration properties_.

ppatierno commented on documentation/api/io.strimzi.api.kafka.model.KafkaAuthorizationOpa.adoc:
```diff
@@ -1,30 +1,30 @@
-To use link:https://www.openpolicyagent.org/[Open Policy Agent^] authorization, set the `type` property in the `authorization` section to the value `opa`.
-The Open Policy Agent authorizer has several configuration options:
+To use link:https://www.openpolicyagent.org/[Open Policy Agent^] authorization, set the `type` property in the `authorization` section to the value `opa`,
```
as far as I understood these changes on OPA related stuff are not related to the PR purpose. Can we do that in a different PR?

PaulRMellor replied:
They're just a few formatting changes to make consistent with the other descriptions added to the appendix. So it's sort of related.

scholzj commented:
@PaulRMellor Could you please look at the conflicts:

PaulRMellor commented:
@scholzj After the merge conflict fix, I fixed a few broken links for the template updates. 
I also changed the format of the template KafkaContainer config, as it wasn't presenting correctly

16f0781caee677a0181184b482eb2f9b255e6c70
